### DISCUSSION TITLE
add rule to prefer double quotes intead of single (like black)

### DISCUSSION THREAD

**Discussion by ssbarnea** [ansible/ansible-lint#584] [Ideas]
In order to avoid discussion about code style, it would be a good idea to add a new rule that requires use of double quotes for escaping strings instead of single quotes, similar to what black does -- this should avoid the issue of mixed quoting.

Because this rule is likely to upset few users, it should be optional.

**Comment by Snelius30** [comment]
It should be rules which allow to set double quote to strings being interpolated and single quotes other cases.

**Comment by greg-hellings** [comment]
Please definitely make this optional, as I prefer no quotes unless YAML requires it for unambiguous strings (e.g. strings that begin with Jinja templating, and that's about it). One of the best things about YAML is that I don't need to quote string values in dictionaries unless there would be ambiguity about the syntax of the file.

**Reply by ssbarnea** [reply]
You are right we want to keep the YAML as simple as possible, we will not add quotes unless they are required. In fact we already prefer double quotes like black but I guess that the OP was confused about the fact that jinja comes inside a quoted string, so that is why we are likely to see single strings inside it.

Is not that we prefer single quotes, it is a side effect of using the double ones for the entire string. That is exactly what black does too.

AFAIK, I do not see any change being needed here.

**Reply by Snelius30** [reply]
Without quotes some kind of values may be interpreted differently. 
Take a look YAML docs
`Quoted styles are useful when a string starts or end with one or more relevant spaces, because unquoted strings are trimmed on both end when parsing their contents. Quotes are required when the string contains special or reserved characters.`
I want to provide a rule to the quotes requirement. It should be optional of course (no restriction by default). It's may be useful for some cases i want to provide consistent code style i.e. Single quotes for "static" strings and double quotes for the interpolated or double quotes only mode.

**Comment by nre-ableton** [comment]
Though I also prefer double-quoted strings, it should be noted that buried in [Ansible's Windows usage documentation](https://docs.ansible.com/ansible/latest/os_guide/windows_usage.html#yaml-style) is this strictly-worded YAML style recommendation:

> You should only quote strings when it is absolutely necessary or required by YAML, and then use single quotes.

If anything, there should be a rule to enforce single-quoted strings, and non-quoted strings by default. Also, I filed another feature request for ansible-lint to have a rule about quoted style in general:

- https://github.com/ansible/ansible-lint/discussions/3937
